main.go

```diff
@@ -40,11 +47,20 @@ func main() {
 			break
 		}
 
-		if re.Match(data) {
-			if !bytes.HasSuffix(data, []byte{'\n'}) {
-				data = append(data, '\n')
+		for _, i := range flagTest {
+			re, regErr := regexp.Compile(i)
+			if regErr != nil {
+				fmt.Fprintln(os.Stderr, regErr)
+				os.Exit(1)
+			}
+
+			if re.Match(data) {
+				if !bytes.HasSuffix(data, []byte{'\n'}) {
+					data = append(data, '\n')
+				}
+				_, _ = os.Stdout.Write(data)
+				break
 			}
```

Comment:
You're compiling the regexes per **every single line**.

Reply:
i could save my re's into an array before the big for loop but i dont know if its possible to use it like i do at the moment. for example if i got arr := []*Regexp{re, re1, re2} can i then do something like that? -> arr[0].Match(data)?

Reply:
Try it out.

Reply:
if i cant i have absolutely no idea how to fix it :D